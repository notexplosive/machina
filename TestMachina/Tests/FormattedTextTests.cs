using ApprovalTests;
using ApprovalTests.Reporters;
using FluentAssertions;
using Machina.Data.TextRendering;
using Microsoft.Xna.Framework;
using TestMachina.Utility;
using Xunit;

namespace TestMachina.Tests
{
    [UseReporter(typeof(DiffReporter))]
    public class FormattedTextTests
    {
        [Fact]
        public void formatted_text_gets_right_number_of_tokens()
        {
            var font1 = new MonospacedFontMetrics(new Point(4, 4));
            var font2 = new MonospacedFontMetrics(new Point(4, 5));

            var subject = new FormattedText(
                new FormattedTextFragment("Hello in blue. ", font1, Color.Blue),
                new FormattedTextFragment("Hello in red!", font2, Color.Red)
                );

            subject.FormattedTokens().Should().HaveCount(11);
        }

        [Fact]
        public void formatted_text_can_render_itself()
        {
            var font1 = new MonospacedFontMetrics(new Point(4, 4));
            var font2 = new MonospacedFontMetrics(new Point(4, 5));

            var subject = new FormattedText(
                new FormattedTextFragment("Hello in blue. ", font1, Color.Blue),
                new FormattedTextFragment("Hello in red!", font2, Color.Red)
                );

            var renderedText = subject.GetRenderedText(Point.Zero);

            Approvals.Verify(TextMeasureUtils.DrawRenderedText(new AsciiDrawPanel(new Point(113, 5)), renderedText));
        }
    }
}