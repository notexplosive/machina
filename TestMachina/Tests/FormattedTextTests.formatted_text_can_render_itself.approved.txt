................................................................................................................ 
.  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  . 
. H.. e.. l.. l.. o..  .. i.. n..  .. b.. l.. u.. e.. ...  .. H.. e.. l.. l.. o..  .. i.. n..  .. r.. e.. d.. !. 
.............................................................  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  . 
                                                            .................................................... 
